src/Microsoft.AspNet.Mvc.Razor/Internal/RouteValueUtility.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+
+namespace Microsoft.AspNet.Mvc.Razor.Internal
+{
+    /// <summary>
+    /// Utility for reading case-normalized route values.
+    /// </summary>
+    public class RouteValueUtility
+    {
+        /// <summary>
+        /// Gets the case-normalized route value for the specified route <paramref name="key"/>.
+        /// </summary>
+        /// <param name="context">The <see cref="ActionContext"/>.</param>
+        /// <param name="key">The route key to lookup.</param>
+        /// <returns>The value corresponding to the key.</returns>
+        /// <remarks>
+        /// The casing of a route value in <see cref="ActionContext.RouteData"/> is determined by the client.
+        /// This making constructing paths for view locations in a case sensitive file system unreliable. Using the
+        /// <see cref="ActionDescriptor.RouteValueDefaults"/> for attribute routes and
+        /// <see cref="ActionDescriptor.RouteConstraints"/> for traditional routes to get route values produces
+        /// consistently cased results.
+        /// </remarks>
+        public static string GetNormalizedRouteValue(ActionContext context, string key)
+        {
+            var actionDescriptor = context.ActionDescriptor;
+            if (actionDescriptor.AttributeRouteInfo != null)
+            {
+                // For attribute routes, use an inline constraint if available.
+                object match;
+                if (actionDescriptor.RouteValueDefaults.TryGetValue(key, out match))
+                {
+                    return match.ToString();
```

Comment:
Should we verify that this isn't doing more than changing the case? For instance if the use manually edits route values (wierd idea, but someone will try it).